src/shared/nldef.rs

```diff
@@ -0,0 +1,204 @@
+// Licensed under the Apache License, Version 2.0
+// <LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// All files in the project carrying such notice may not be copied, modified, or distributed
+// except according to those terms.
+use shared::basetsd::ULONG64;
+use shared::minwindef::ULONG;
+use shared::ntdef::BOOLEAN;
+ENUM!{enum NL_PREFIX_ORIGIN {
+    IpPrefixOriginOther = 0,
+    IpPrefixOriginManual = 1,
+    IpPrefixOriginWellKnown = 2,
+    IpPrefixOriginDhcp = 3,
+    IpPrefixOriginRouterAdvertisement = 4,
+    IpPrefixOriginUnchanged = 1 << 4,
+}}
+pub const NlpoOther: NL_PREFIX_ORIGIN = IpPrefixOriginOther;
+pub const NlpoManual: NL_PREFIX_ORIGIN = IpPrefixOriginManual;
+pub const NlpoWellKnown: NL_PREFIX_ORIGIN = IpPrefixOriginWellKnown;
+pub const NlpoDhcp: NL_PREFIX_ORIGIN = IpPrefixOriginDhcp;
+pub const NlpoRouterAdvertisement: NL_PREFIX_ORIGIN = IpPrefixOriginRouterAdvertisement;
+ENUM!{enum NL_SUFFIX_ORIGIN {
+    IpSuffixOriginOther = 0,
+    IpSuffixOriginManual = 1,
+    IpSuffixOriginWellKnown = 2,
+    IpSuffixOriginDhcp = 3,
+    IpSuffixOriginLinkLayerAddress = 4,
+    IpSuffixOriginRandom = 5,
+    IpSuffixOriginUnchanged = 1 << 4,
+}}
+pub const NlsoOther: NL_SUFFIX_ORIGIN = IpSuffixOriginOther;
+pub const NlsoManual: NL_SUFFIX_ORIGIN = IpSuffixOriginManual;
+pub const NlsoWellKnown: NL_SUFFIX_ORIGIN = IpSuffixOriginWellKnown;
+pub const NlsoDhcp: NL_SUFFIX_ORIGIN = IpSuffixOriginDhcp;
+pub const NlsoLinkLayerAddress: NL_SUFFIX_ORIGIN = IpSuffixOriginLinkLayerAddress;
+pub const NlsoRandom: NL_SUFFIX_ORIGIN = IpSuffixOriginRandom;
```

Comment:
```suggestion
ENUM!{enum NL_SUFFIX_ORIGIN {
    NlsoOther = 0,
    NlsoManual,
    NlsoWellKnown,
    NlsoDhcp,
    NlsoLinkLayerAddress,
    NlsoRandom,
    IpSuffixOriginOther = 0,
    IpSuffixOriginManual,
    IpSuffixOriginWellKnown,
    IpSuffixOriginDhcp,
    IpSuffixOriginLinkLayerAddress,
    IpSuffixOriginRandom,
    IpSuffixOriginUnchanged = 1 << 4,
}}
```
Make sure to match what the headers do.